index.js

```diff
@@ -552,90 +551,73 @@ prototype.buildParameter = function(oldParameter) {
 };
 
 /*
- * Convertes Swagger 1.2 authorization definitions to Swagger 2.0 security
- *   definitions
- *
- * @param resourceListing {object} - The Swagger 1.2 Resource Listing document
- * @param convertedSecurityNames {object} - A list of original Swagger 1.2
- * authorization names and the new Swagger 2.0
- *  security names associated with it (This is required because Swagger 2.0 only
- *  supports one oauth2 flow per security definition but in Swagger 1.2 you
- *  could describe two (implicit and authorization_code).  To support this, we
- *  will create a per-flow version of each oauth2 definition, where necessary,
- *  and keep track of the new names so that when we handle security references
- *  we reference things properly.)
+ * Convertes Swagger 1.2 authorization definitions into Swagger 2.0 definitions
+ * Definitions couldn't be converted 1 to 1, 'this.securityNamesMap' should be
+ * used to map between Swagger 1.2 names and one or more Swagger 2.0 names.
  *
+ * @param oldAuthorizations {object} - The Swagger 1.2 Authorizations definitions
  * @returns {object} - Swagger 2.0 security definitions
  */
-prototype.buildSecurityDefinitions = function(resourceListing,
-  convertedSecurityNames) {
-  if (isEmpty(resourceListing.authorizations)) {
-    return undefined;
-  }
-
+prototype.buildSecurityDefinitions = function(oldAuthorizations) {
   var securityDefinitions = {};
 
-  Object.keys(resourceListing.authorizations).forEach(function(name) {
-    var authorization = resourceListing.authorizations[name];
-    var createDefinition = function createDefinition(oName) {
-      var securityDefinition = securityDefinitions[oName || name] = {
-        type: authorization.type
-      };
+  this.securityNamesMap = {};
+  this.forEach(oldAuthorizations, function(oldAuthorization, name) {
+    var scopes = {};
+    this.forEach(oldAuthorization.scopes, function(oldScope) {
```

Comment:
Yes because of this [line](https://github.com/apigee-127/swagger-converter/blob/master/index.js#L723), basically `forEach` allow you to forget about passing `this`.